forc/src/cli/commands/test.rs

```diff
@@ -110,13 +109,25 @@ fn print_tested_pkg(pkg: &TestedPackage, test_print_opts: &TestPrintOpts) -> Res
     };
     if failed != 0 {
         info!("\n   failures:");
-        for (failed_test_name, failed_test_detail) in failed_test_details {
-            let path = &*failed_test_detail.file_path;
-            let line_number = failed_test_detail.line_number;
+        for failed_test in failed_tests {
+            let failed_test_name = &failed_test.name;
+            let failed_test_details = failed_test.details()?;
+            let path = &*failed_test_details.file_path;
+            let line_number = failed_test_details.line_number;
+            let revert_code = failed_test
+                .revert_code()
+                .ok_or_else(|| anyhow::anyhow!("missing revert code for failed test"))?;
```

Comment:
Does a test that unexpectedly returns successfully when decorated with `should_revert` always produce a revert code?

Reply:
Thats a good question I changed my approach so that I only print out details if we have a revert code present.